tests/test_physical_systems/test_electric_motors.py

```diff
@@ -0,0 +1,253 @@
+     assert concreteDCMotor.nominal_values == test_dcmotor_nominal_values
+     assert concreteDCMotor.limits == test_dcmotor_limits
+     assert concreteDCMotor._initial_states == {"i_a": 10.0, "i_e": 15.0}
+
+def test_DCMotor_torque(concreteDCMotor):
+     
+     currents = [100, 10, 2, 3]
+     expectedtorque = concreteDCMotor.motor_parameter["l_e_prime"]* currents[0] * currents[1]
+
+     assert concreteDCMotor.torque(currents) == expectedtorque
+
+def test_DCMotor_i_in(concreteDCMotor):
+     
+     currents = [100, 10, 2, 3]
+     expectedcurrent = currents
+
+     assert concreteDCMotor.i_in(currents) == expectedcurrent
+
+def test_DCMotor_electrical_ode(concreteDCMotor):
+     
+     concreteDCMotor._model_constants[0] = concreteDCMotor._model_constants[0] / concreteDCMotor.motor_parameter["l_a"]
+     concreteDCMotor._model_constants[1] = concreteDCMotor._model_constants[1] / concreteDCMotor.motor_parameter["l_e"]
+     u_in = [10,20]
+     omega = 60
+     state = [10.0,15.0]
+
+     expectedElectricalOde = np.matmul(
+            concreteDCMotor._model_constants,
+            np.array(
+                [
+                    state[0],
+                    state[1],
+                    omega * state[1],
+                    u_in[0],
+                    u_in[1],
+                ]
+            ),
+        )
+     
+     assert np.array_equal(concreteDCMotor.electrical_ode(state,u_in,omega), expectedElectricalOde)
+
+def test_DCMotor_get_state_space(concreteDCMotor):
+     
+    inputvoltages = Box(0, 1, shape=(2,), dtype=np.float64)
+    inputscurrents = Box(-1, 1, shape=(2,), dtype=np.float64)
+
+    expectedlow = {
+            "omega":0,
+            "torque": -1,
+            "i_a": -1 ,
+            "i_e": -1 ,
+            "u_a": 0,
+            "u_e":  0,
+        }
+    
+    expectedhigh = {"omega": 1, "torque": 1, "i_a": 1, "i_e": 1, "u_a": 1, "u_e": 1}
+    
+    assert concreteDCMotor.get_state_space(inputscurrents,inputvoltages)[0] == expectedlow
+    assert concreteDCMotor.get_state_space(inputscurrents,inputvoltages)[1] == expectedhigh
+
+def test_ConcreteDCMotor_reset(concreteDCMotor):
+    new_initial_state = {'i_a': 100.0, 'i_e': 20.0}
+    default_initial_state = {"i_a": 10.0, "i_e": 15.0}
+    default_Initial_state_array = [10.0, 15.0]
+    extex_state_positions = {
+          "omega": 0,
+          "torque": 1,
+          "i":2,
+          "i_a":3,
+          "i_e": 4,
+          "u_a": 5,
+          "u_e": 6,
+    }
+     
+    extex_state_space = Box(low=-1, high=1, shape=(7,), dtype=np.float64)
+    assert concreteDCMotor._initial_states == default_initial_state
+    concreteDCMotor._initial_states = {'i_a': 100.0, 'i_e': 20.0}
+
+    assert concreteDCMotor._initial_states == new_initial_state
+    assert np.array_equal(concreteDCMotor.reset(extex_state_space,extex_state_positions),default_Initial_state_array)
+    assert concreteDCMotor._initial_states ==  default_initial_state
```

Comment:
Could you also test the behavior when concreteDCMotor._initializer["states"] is overwritten? In that case we assume the new initial values will not be overwritten

Reply:
This is what I have already tried to do with "test_ConcreteDCMotor_reset", first the motor is initialized with a state different from the one in DCMotor.py and then the state is again changed with the "new_initial_state". On reset the intial_State is changed to default_initial_state which is different from the value in the DCMotor.py. I hope this is what you expect. Thank you!